### PR TITLE
Various changes to Dockerfile

### PR DESCRIPTION
**Pull request by robotblake**
- Use the "python:2.7-alpine" base image
- Put application in /app as per "Dockerflow"
- Add script that uses $PORT and $WORKERS when starting gunicorn
- Re-order steps to take advantage of caching

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,22 +1,19 @@
-# ubuntu 16.04 xenial image
-FROM ubuntu
+# use base python
+FROM python:2.7
 
-# apt update it
-RUN apt-get update
-# get essentials
-RUN apt-get install -y tar git curl wget dialog net-tools build-essential python python-dev python-distribute python-pip
```
Comment (maurodoglio) [inline Dockerfile:7]: does the python:2.7 image come with all these packages by default? I would be surprised if it shipped with git by default.

File: bin/start [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+set -eo pipefail
+
+: ${WORKERS:=4}
```
Comment (maurodoglio) [inline bin/start:5]: I'm not familiar with this syntax, what does it do?

File: Dockerfile [hunk 3]
```diff
@@ -1,22 +1,19 @@
...
-RUN pip install -r /telemetry-schema-service/requirements.txt
-
-# exposing 8080 for flask app
+# expose 8080 for flask app
 EXPOSE 8080
 
-# workdir
-WORKDIR /telemetry-schema-service/app
+# copy application
+COPY . /app/
```
Comment (maurodoglio) [inline Dockerfile:19]: I'm not sure why you copied the requirements file separately.
I would suggest to move this line to line 9

Comment (robotblake) [inline Dockerfile:19]: To take advantage of docker layer caching, this way if the requirements.txt doesn't change it doesn't need to re-download all of those packages.

https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/add-or-copy

File: bin/start (hunk 2 above)
Comment (robotblake) [inline bin/start:5]: The `:` is a bash built-in that lets the rest of the line work but is itself a no-op, the `${blah:="something"}` syntax sets a variable if it's not already set.  There's a decent explanation on [StackExchange](http://unix.stackexchange.com/questions/25425/what-does-param-value-mean#answer-25434).

tl;dr: Sets the variable if it hasn't been set yet.

File: Dockerfile (hunk 1 above)
Comment (robotblake) [inline Dockerfile:7]: It comes with tar, git, curl, and wget.  I'm not sure why we needed dialog or net-tools here, maybe @purukaushik knows?  None of the packages in requirements need build-essential / a compiler and it severely bloats the image size.  Python and Pip are compiled / installed in the base image.

```
root@5af2fe28c5d0:/app# dpkg -l | grep "^ii  \(tar\|git\|curl\|wget\|dialog\|net-tools\) "
ii  curl                              7.38.0-4+deb8u3            amd64        command line tool for transferring data with URL syntax
ii  git                               1:2.1.4-2.1+deb8u2         amd64        fast, scalable, distributed revision control system
ii  tar                               1.27.1-2+b1                amd64        GNU version of the tar archiving utility
ii  wget                              1.16-1                     amd64        retrieves files from the web
root@5af2fe28c5d0:/app# python --version
Python 2.7.12
root@5af2fe28c5d0:/app# pip --version
pip 8.1.2 from /usr/local/lib/python2.7/site-packages (python 2.7)
```

File: bin/start (hunk 2 above)
Comment (maurodoglio) [inline bin/start:5]: Thanks for the clarification :)

File: Dockerfile (hunk 1 above)
Comment (maurodoglio) [inline Dockerfile:7]: wow. I really thought that image was kind of minimal. @purukaushik can you confirm the app doesn't need those extra packages?

File: Dockerfile (hunk 3 above)
Comment (maurodoglio) [inline Dockerfile:19]: oh I see now

Comment (robotblake) [inline Dockerfile:19]: Meta point, maybe we should generate the requirements.txt with `pip freeze` so it has version info? Doesn't have to be part of this PR though.

File: Dockerfile (hunk 1 above)
Comment (purukaushik) [inline Dockerfile:7]: I can confirm that we don't need those. I was following a docker example and had it there for a while. All the stuff needed by the app is in the python requirements.txt file. Also since the tests passed, we're good in removing the apt-get installs.
